test/tests/query_params_test.js

```diff
@@ -3,8 +3,24 @@ import Router from "router";
 import { resolve, configure, reject, Promise } from "rsvp";
 
 var router, url, handlers, expectedUrl, actions;
+var scenarios = [
+  {
+    name: 'Sync Get Handler',
+    getHandler: function(name) {
+      return handlers[name] || (handlers[name] = {});
+    }
+  },
+  {
+    name: 'Async Get Handler',
+    getHandler: function(name) {
+      return Promise.resolve(handlers[name] || (handlers[name] = {}));
```

Comment:
To ensure we are async (and not just chaining on a `.then`able synchronously) can you change to:

```js
return new Promise(function(resolve) {
  setTimeout(function() {
    resolve(handlers[name] || (handlers[name] = {}));
  }, 1);
});
```